pkg/registry/apis/userstorage/storage.go

```diff
@@ -0,0 +1,42 @@
+package userstorage
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+
+	userstorage "github.com/grafana/grafana/pkg/apis/userstorage/v0alpha1"
+	grafanaregistry "github.com/grafana/grafana/pkg/apiserver/registry/generic"
+	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
+)
+
+var _ grafanarest.Storage = (*storage)(nil)
+
+type storage struct {
+	*genericregistry.Store
+}
+
+func newStorage(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter) (*storage, error) {
+	resourceInfo := userstorage.UserStorageResourceInfo
+	strategy := grafanaregistry.NewStrategy(scheme, resourceInfo.GroupVersion())
+	createStrategy := newStrategy(scheme, resourceInfo.GroupVersion())
+
+	store := &genericregistry.Store{
+		NewFunc:                   resourceInfo.NewFunc,
+		NewListFunc:               resourceInfo.NewListFunc,
+		KeyRootFunc:               grafanaregistry.KeyRootFunc(resourceInfo.GroupResource()),
+		KeyFunc:                   grafanaregistry.NamespaceKeyFunc(resourceInfo.GroupResource()),
+		PredicateFunc:             grafanaregistry.Matcher,
+		DefaultQualifiedResource:  resourceInfo.GroupResource(),
+		SingularQualifiedResource: resourceInfo.SingularGroupResource(),
+		TableConvertor:            resourceInfo.TableConverter(),
+		CreateStrategy:            createStrategy,
+		UpdateStrategy:            strategy,
```

Comment:
NIT – I would consider adding some checks on the sizes of individual fields, maybe limiting each value in the map to a certain max length.

@toddtreece WDYT?

Reply:
i don't have a strong opinion here. maybe we could start with logs or metrics to observe the size of objects and/or keys?